org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/DocumentHighlightTest.xtend

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ * Copyright (c) 2016 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ide.tests.server
+
+import io.typefox.lsapi.DocumentHighlight
+import io.typefox.lsapi.DocumentHighlightKind
+import io.typefox.lsapi.Position
+import org.eclipse.xtext.ide.server.occurrences.IDocumentHighlightService
+import org.junit.Test
+
+/**
+ * Class for testing the {@link DocumentHighlightService document highlight
+ * service} behavior through the language server.
+ * 
+ * <p>
+ * The string representation of on single {@link DocumentHighlight highlight}
+ * instance is represented with the below pattern:
+ * 
+ * <pre>
+ * R|W|T [{d}+, {d}+] .. [{d}+, {d}+]
+ * </pre>
+ * 
+ * where the first character describes the {@link DocumentHighlightKind kind} of
+ * the document highlight. If missing, then the value is {@code NaN}. The values
+ * between the first square brackets are the start {@link Position#getLine()
+ * line} and the (zero-based) {@link Position#getCharacter() character offset}
+ * in the line. The second square brackets contains the end position of the
+ * selection and described with the same semantics as the first start range.
+ * Multiple highlight instances are separated by the {@code |} ("pipe")
+ * character.
+ * 
+ * @author akos.kitta - Initial contribution and API
+ * 
+ * @see DocumentHighlight
+ * @see IDocumentHighlightService
+ */
+public class DocumentHighlightTest extends AbstractTestLangLanguageServerTest {
+
+	@Test
+	def void singleLineNoOccurrences_SelectionBeforeWriteSiteMethodName() {
+		testDocumentHighlight[
+			model = '''type A { op foo() { } }''';
```

Comment:
You could derive the line / column information directly from the text using a special character, e.g. `'''type A { op# foo() { } }'''` would select offset 11 in the text above (the `#` would be filtered out of the text).